Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -116,6 +127,14 @@ void UPSWorldSubsystem::RegisterSpotComponent(UPSSpotComponent* MySpotComponent)
 		return;
 	}
 	PSSpotComponentArrayInternal.AddUnique(MySpotComponent);
+
+	for (TTuple<FName, FPSRowData>& RowData : ProgressionSettingsDataInternal)
+	{
+		if (RowData.Value.Character == MySpotComponent->GetMeshChecked().GetPlayerTag())
+		{
+			PSSpotTagArrayInternal.Add(RowData.Key, MySpotComponent);
```

Comment:
Please also call `Empty()` for your tmap in `PerformCleanUp()` to avoid memory leaks